### PR TITLE
nixos/modules/services/x11/hardware/: add cmt module 

### PR DESCRIPTION
**Pull request by kcalvinalvin**
###### Motivation for this change
To add support for cmt (Chrome Multitouch), custom built drivers for Chromebooks.

Known issues: Can't control such things as acceleration through the desktop environment's GUI. Places an empty 50 or 60 .conf file in the /etc/X11/xorg.conf.d/

###### Things done
Fixes applied for the changes requested for #66195 (git commit cleanup and various changes)

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @Infinisil

### CODE REVIEW

**Comment by kcalvinalvin** [comment]
I've updated the maintainer-list.nix file to include me but not sure how to start the tests again.

**Comment by alyssais** [comment]
Can you fix all the trailing whitespace please?

Also, I think you might have misunderstood how licenses.nix. The license
in this case appears to just be bsd3, so you can just use that unless
there's something extra specific to ChromiumOS I'm missing?

Finally, commit messages should be in the form described in the Nixpkgs
manual.

**Comment by kcalvinalvin** [comment]
Fixed trailing whitespace and commit message. I'm still not sure if this is the right commit message to have since there are no examples for making new modules so please let me know if it's not following the formatting.

For ChromiumOS there needs to be [this](https://github.com/kcalvinalvin/ChromiumOS_License/blob/master/LICENSE) text included.

I wasn't quite sure how to go about it so I just made a repo for it but it seems like github just added BSD3 to the repo. If there is a better way of going about it, I'll certainly do so.

**Comment by alyssais** [comment]
> Fixed trailing whitespace and commit message. I'm still not sure if
> this is the right commit message to have since there are no examples
> for making new modules so please let me know if it's not following the
> formatting.

Since you're adding new packages and new modules in the same commit, I
think it would be fine to just phrase the commit message as if you were
adding the package.

> For ChromiumOS there needs to be
> [this](https://github.com/kcalvinalvin/ChromiumOS_License/blob/master/LICENSE)
> text included.
>
> I wasn't quite sure how to go about it so I just made a repo for it
> but it seems like github just added BSD3 to the repo. If there is a
> better way of going about it, I'll certainly do so.

This is just the standard 3-Clause BSD license, with appropriate
substitutions made for the names of the project and authors. So we can
just use licenses.bsd3. Otherwise we'd end up with an entry in
licenses.nix for every project and it would be tens of thousands of
lines long.

**Comment by kcalvinalvin** [comment]
Changed commit message and removed/replaced chromeos license to bsd3

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3181,6 +3181,12 @@
     githubId = 1047859;
     name = "Kaz Wesley";
   };
+  kcalvinalvin = {
+    email = "calvin@kcalvinalvin.info";
+    github = "kcalvinalvin";
+    githubId = "37185887";
```
Comment (Infinisil) [inline maintainers/maintainer-list.nix]: Should be an integer, not a string

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    services.xserver.cmt = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable chrome multitouch input (cmt). Touchpad drivers that are configured for chromebooks.";
+      };
+      models = mkOption {
+        type = types.enum [ "atlas" "banjo" "candy" "caroline" "cave" "celes" "clapper" "cyan" "daisy" "elan" "elm" "enguarde" "eve" "expresso" "falco" "gandof" "glimmer" "gnawty" "heli" "kevin" "kip" "leon" "lulu" "orco" "pbody" "peppy" "pi" "pit" "puppy" "quawks" "rambi" "samus" "snappy" "spring" "squawks" "swanky" "winky" "wolf" "auron_paine" "auron_yuna" "daisy_skate" "nyan_big" "nyan_blaze" "veyron_jaq" "veyron_jerry" "veyron_mighty" "veyron_minnie" "veyron_speedy" ];
+        default = [];
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: This isn't a valid default, you should remove this

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        example = [ "banjo" ];
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: Not a valid example either, the type doesn't represent a list

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        description = "Which models to enable cmt for. Enter the Code Name for your chromebook model. Code name can be found at www.chromium.org/chromium-os/developer-information-for-chrome-os-devices.";
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: You need `<link xlink:href="https://example.com"/>` for links

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      };
+      "${etcPath}/50-touchpad-cmt-${cfg.models}.conf" = {
+        source = "${pkgs.chromium-xorg-conf}/50-touchpad-cmt-${cfg.models}.conf";
+      };
+      "${etcPath}/60-touchpad-cmt-${cfg.models}.conf" = {
+        source = "${pkgs.chromium-xorg-conf}/60-touchpad-cmt-${cfg.models}.conf";
+      };
+    };
+
+  assertions = [
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: Indentation not correct

**Review by infinisil — CHANGES_REQUESTED** [review]

File: nixos/modules/services/x11/hardware/cmt.nix (hunk 5 above)
Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: fixed

File: nixos/modules/services/x11/hardware/cmt.nix (hunk 4 above)
Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: fixed

File: nixos/modules/services/x11/hardware/cmt.nix (hunk 3 above)
Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: fixed

File: nixos/modules/services/x11/hardware/cmt.nix (hunk 2 above)
Comment (kcalvinalvin) [reply nixos/modules/services/x11/hardware/cmt.nix]: removed

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (kcalvinalvin) [reply maintainers/maintainer-list.nix]: fixed

**Comment by Infinisil** [comment]
@GrahamcOfBorg build libgestures
@GrahamcOfBorg build libevdevc
@GrahamcOfBorg build xf86_input_cmt
@GrahamcOfBorg build chromium-xorg-conf

**Comment by kcalvinalvin** [comment]
Is there anything else that I need to do?

**Comment by Infinisil** [comment]
Would be nice if you could clean up the commits. Initial commit for adding to the maintainer list, then one commit for each package, then a commit for the NixOS module

**Comment by kcalvinalvin** [comment]
Cleaned up the commits as requested

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Thanks, LGTM :)

**Pull request merged**